src/components/RadioButtonGroup/RadioButtonGroup.tsx

```diff
@@ -49,25 +66,41 @@ const RadioButtonGroupComponent = (
     value: activeValue,
     name: customName,
     label,
+    invalid,
+    validationHint,
+    showValid,
+    disabled,
+    hasWarning,
     ...props
   }: RadioButtonGroupProps,
   ref: RadioButtonGroupProps['ref'],
 ) => {
   const name = customName || uniqueId('radio-button-group_');
   return (
-    <div role="group" aria-label={label} ref={ref} {...props}>
-      {options &&
-        options.map(({ children, value, className, ...rest }) => (
-          <div key={value && value.toString()} className={className}>
-            <RadioButton
-              {...{ ...rest, value, name, onChange }}
-              checked={value === activeValue}
-            >
-              {children}
-            </RadioButton>
-          </div>
-        ))}
-    </div>
+    <fieldset name={name}>
+      {label && <legend>{label}</legend>}
+
+      <div role="group" aria-label={label} ref={ref} {...props}>
```

Comment:
Now that the radio buttons are wrapped inside the fieldset, this div has become obsolete. Please remove it and move the `ref` and `{...props}` props to the fieldset element. The `role` and `aria-label` props are no longer needed.